### PR TITLE
Add input handlers guide

### PR DESCRIPTION
**Pull request by FichteFoll**
Here's what I have been working on on and off for 2 weeks (I think). I have compiled all currently known knowledge about input handlers into a guide page that explains what they do and how they work, as well as mention a few quirks. 

Some input methods (`confirm`, `cancel`, `initial_text`) have been omitted because I couldn't think of an easy and feasible use case that could fit into my examples and think their documentation is clear enough.

Generally this is the kind of style I intend to go for with topics like this inside the guide section. Any feedback is greatly appreciated.

I will merge this if no reviews have been submitted in 7 days.

### CODE REVIEW

File: docs/guide/extensibility/plugins/input_handlers.md [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+and explain more gears you can tweak
+for advanced configuration.
+
+::: warning Important
+To use an input handler for a command,
+**the command must have an entry in the [Command Palette][]**.
+This is easy to forget,
+so make sure to remember!
+
+[Command Palette]: ../comman_palette.md
```
Comment (keith-hall) [inline docs/guide/extensibility/plugins/input_handlers.md:62]: typo?
```suggestion
[Command Palette]: ../command_palette.md
```

File: docs/guide/extensibility/plugins/input_handlers.md [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+since we need two arguments,
+and information about the previous input handler is lost.
+
+::: tip
+Having problems running this command?
+Did you add a `.sublime-commands` entry for it?
+:::
+
+
+### The `input_next` Method
```
Comment (Ultra-Instinct-05) [inline docs/guide/extensibility/plugins/input_handlers.md:396]: Should be `next_input` & not `input_next`
```suggestion
### The `next_input` Method
```

File: docs/guide/extensibility/plugins/input_handlers.md [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+
+The `preview` method is called
+for every modification of the entered text
+and allows to show a small preview
+below the Command Palette.
+The preview can either be pure text
+or can use [minihtml][]
+for a markup-enabled format.
+
+[minihtml]: https://www.sublimetext.com/docs/3/minihtml.html
```
Comment (jrappen) [inline docs/guide/extensibility/plugins/input_handlers.md:181]: target new docs

File: docs/guide/extensibility/plugins/input_handlers.md [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+- A command's `input` method may be called multiple times
+  until the user can access it.
+
+- `is_visible` and `is_enabled` cannot decide their return value
+  based on the given arguments
+  when an input handler is involved.
+  ([#3249](https://github.com/sublimehq/sublime_text/issues/3249))
+
+
+[api-TextCommand]: https://www.sublimetext.com/docs/3/api_reference.html#sublime_plugin.TextCommand
```
Comment (jrappen) [inline docs/guide/extensibility/plugins/input_handlers.md:503]: target new docs

File: docs/guide/extensibility/plugins/input_handlers.md [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+[api-TextInputHandler]: https://www.sublimetext.com/docs/3/api_reference.html#sublime_plugin.TextInputHandler
```
Comment (jrappen) [inline docs/guide/extensibility/plugins/input_handlers.md:504]: target new docs

File: docs/guide/extensibility/syntaxdefs.md [hunk 6]
```diff
@@ -28,7 +28,7 @@ Documentation is available
 at the [official documentation][].
 :::
 
-[official documentation]: https://www.sublimetext.com/docs/3/syntax.html)
+[official documentation]: https://www.sublimetext.com/docs/3/syntax.html
```
Comment (jrappen) [inline docs/guide/extensibility/syntaxdefs.md:31]: target new docs

File: docs/guide/extensibility/plugins/input_handlers.md (hunk 3 above)
Comment (FichteFoll) [reply docs/guide/extensibility/plugins/input_handlers.md:181]: One comment would have been enough.

Alas, no, I'm not referencing the new docs until they have been made public. I will, however, update all urls as soon as they are public, which will be a simple `sed` call.

**Comment by FichteFoll** [comment]
I switched the branch to use git LFS for the mp4 files as an experiment. Let's see if Actions can handle that natively or whether I have to update the workflow.

**Pull request merged**